hydra-redis/redisgraph_demo1.py

```diff
@@ -0,0 +1,294 @@
+    """it makes a node for every collection endpoint.
+    """
+    print("accessing every collection in entrypoint")
+    for obj in collection_endpoint:
+        obj_method = []
+        properties = []
+        obj_properties = {}
+        print(
+            "check url for endpoint",
+            base_url +
+            collection_endpoint[obj])
+        new_url = base_url + collection_endpoint[obj]
+        for obj2 in api_doc.collections[obj]["collection"].supportedOperation:
+            obj_method.append(obj2.method)
+        obj_properties["methods"] = str(obj_method)
+        print(obj_properties["methods"])
+        for obj2 in api_doc.collections[obj]["collection"].supportedProperty:
+            properties.append(str(obj2.desc[4:]))
+        obj_properties["title"] = str(properties)
+        response = ur.urlopen(base_url + collection_endpoint[obj])
+        new_file = json.loads(response.read().decode('utf-8'))
+        obj_properties["members"] = str(new_file["members"])
+        obj_collection_node = Node(
+            label="id", alias=obj, properties=obj_properties)
+        redis_graph.add_node(obj_collection_node)
+        edge = Edge(url_node, "has_collection", obj_collection_node)
+        redis_graph.add_edge(edge)
+        collectionobjects(
+            obj_collection_node,
+            new_file["members"],
+            new_url,
+            url_node)
+
+
+def get_apistructure(entrypoint_obj, url_node):
+    """ It breaks the endpoint into two parts collection and classes"""
+    collection_endpoint = {}
+    classes_endpoint = {}
+    collection = 0
+    classes = 0
+    print("split entrypoint into 2 types of endpoints collection and classes")
+    for obj in url_node.properties:
+        if obj != "@id" and obj != "@type" and obj != "@context":
+            if obj in api_doc.collections.keys():
+                collection = 1
+                collection_endpoint[obj] = str(url_node.properties[obj])
+            else:
+                classes = 1
+                classes_endpoint[obj] = str(url_node.properties[obj])
+    if collection == 1:
+        endpointCollection(collection_endpoint, url_node)
+
+    if classes == 1:
+        endpointclasses(classes_endpoint, url_node)
+
+
+def get_endpoints(entrypoint_obj):
+    """create node for entrypoint"""
+    print("creating entrypoint node")
+    url_node = Node(label="id", alias="Entrypoint", properties=entrypoint_obj)
+    redis_graph.add_node(url_node)
+    return get_apistructure(entrypoint_obj, url_node)
+
+
+if __name__ == "__main__":
+    url = "http://35.224.198.158:8080/api"
+    match_obj = re.match(r'(.*)://(.*)/(.*)/?', url, re.M | re.I)
+    base_url = "{0}://{1}".format(match_obj.group(1), match_obj.group(2))
+    entrypoint = "/" + match_obj.group(3) + "/"
+    print("baseurl, entrypoint ", base_url, entrypoint)
+
+    apidoc = final_file(url + "/vocab")
+    api_doc = doc_maker.create_doc(apidoc)
+    get_endpoints(api_doc.entrypoint.get())
```

Comment:
You are still not using the abstraction given by the `doc_maker`, when you use the `get()` method, it returns the json object, which is the same as what was being used previously.

Reply:
Sorry, I cannot get your point here. I think, we need only the endpoints ids with its name here and it(object) provides only the same thing.